### PR TITLE
Adding new DGGRID tool package

### PR DESCRIPTION
**Pull request by allixender**
## Adding new DGGRID tool package

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| c/c++           | `@conda-forge/help-c-cpp`     |

Checklist:

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages (proj4 old version, only 2 files functions and statically linked, MIT License), shapelib is configured via cmake to load from host, a vendored version 1.55 is for legacy compile without conda environment)
- [x] If static libraries are linked in, the license of the static library is packaged (MIT, in-file)
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

We are aware of the issue of vendored dependencies, and proj4 is really not comparable (only uses 2 adapted files). Shapelib might be possible to vendor out. We are working together with the original software developer. The software statically links everything into a single binary executable.

### CODE REVIEW

**Comment by allixender** [comment]
Dear Conda Recipe Stage team @conda-forge/help-c-cpp,
I'd hereby like to submit a new binary package.
The build checks went through, I disabled the windows build for now.
The missing check boxes are explained and we hope they should not be show stoppers, as the app is a single binary, and the one-and-a-half potentially vendored libs (the half one is literally only two files from a much older proj version) will not impact the the ecosystem. The shapelib we are engaged with the original app author for whom we want to make the app more accessible for users.

File: recipes/dggrid/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - libgdal
+    - shapelib
+  run:
+    - libgdal
+    - shapelib
+
+test:
+  commands:
+    - test -f ${PREFIX}/bin/dggrid 
+    - if not exist %LIBRARY_BIN%\dggrid exit 1  # [win]
```
Comment (carterbox) [inline recipes/dggrid/meta.yaml:31]: Does ddgrid have a `--version` or `--help` command that you can run to check that the binary is executable in the conda environment?

File: recipes/dggrid/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler("cxx") }}
+    - cmake >=3.12,<4
+    - make  # [linux]
+  host:
+    - libgdal
+    - shapelib
```
Comment (carterbox) [inline recipes/dggrid/meta.yaml:23]: It's good that you are depending on preexisting conda-forge packages, but you need to actually link to them. The logs show that you are not linking to shapelib dynamically.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=652966&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1972

**Review by carterbox — CHANGES_REQUESTED** [review]
> the one-and-a-half potentially vendored libs (the half one is literally only two files from a much older proj version) will not impact the ecosystem.

If you want to revendor libraries, you need to (1) explain for each library why you can't dynamically link to it from another conda-forge package (2) include their licenses in the license section of this recipe.

File: recipes/dggrid/meta.yaml (hunk 2 above)
Comment (allixender) [reply recipes/dggrid/meta.yaml:23]: cmake is now configured to load and dynamically link shapelib from the host, the vendored version 1.55 src is kept for legacy compile without conda environment)

File: recipes/dggrid/meta.yaml (hunk 1 above)
Comment (allixender) [reply recipes/dggrid/meta.yaml:31]: Not currently. The upstream author is informed and said he will add this at some point.
I can offer a minimal test run config:

```shell
echo "dggrid_operation GENERATE_GRID
dggs_type ISEA7H
dggs_res_spec 3
clip_subset_type WHOLE_EARTH
cell_output_type GDAL
cell_output_gdal_format GPKG
cell_output_file_name minout.gpkg" > min.meta && dggrid min.meta
```

This should return ok.

Comment (allixender) [reply recipes/dggrid/meta.yaml:31]: Should this be like a separate script, or can it be added like that to the yaml?

**Comment by carterbox** [comment]
@conda-forge/help-c-cpp, please review

File: recipes/dggrid/meta.yaml (hunk 1 above)
Comment (carterbox) [reply recipes/dggrid/meta.yaml:31]: You can add do either.

**Pull request merged**